integration_tests/create_test.go

```diff
@@ -63,39 +66,77 @@ func TestCreateUserTestSuite(t *testing.T) {
+func LndHubTestServiceInit(usersToCreate int) (svc *service.LndhubService, logins []controllers.CreateUserResponseBody, tokens []string, err error) {
+	// change this if you want to run tests using sqlite
+	// dbUri := "file:data_test.db"
+	//make sure the datbase is empty every time you run the test suite
+	dbUri := "postgresql://user:password@localhost/lndhub?sslmode=disable"
 	c := &service.Config{
-		DatabaseUri:    "file:data_test.db",
+		DatabaseUri:    dbUri,
 		JWTSecret:      []byte("SECRET"),
+		JWTExpiry:      3600,
 		LNDAddress:     "rpc.lnd1.regtest.getalby.com:443",
 		LNDMacaroonHex: "0201036c6e6402f801030a10e2133a1cac2c5b4d56e44e32dc64c8551201301a160a0761646472657373120472656164120577726974651a130a04696e666f120472656164120577726974651a170a08696e766f69636573120472656164120577726974651a210a086d616361726f6f6e120867656e6572617465120472656164120577726974651a160a076d657373616765120472656164120577726974651a170a086f6666636861696e120472656164120577726974651a160a076f6e636861696e120472656164120577726974651a140a057065657273120472656164120577726974651a180a067369676e6572120867656e657261746512047265616400000620c4f9783e0873fa50a2091806f5ebb919c5dc432e33800b401463ada6485df0ed",
 	}
 	dbConn, err := db.Open(c.DatabaseUri)
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to database: %w", err)
+		return nil, nil, nil, fmt.Errorf("failed to connect to database: %w", err)
 	}
 	ctx := context.Background()
 	migrator := migrate.NewMigrator(dbConn, migrations.Migrations)
 	err = migrator.Init(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("failed to init migrations: %w", err)
+		return nil, nil, nil, fmt.Errorf("failed to init migrations: %w", err)
 	}
 	_, err = migrator.Migrate(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("failed to migrate: %w", err)
+		return nil, nil, nil, fmt.Errorf("failed to migrate: %w", err)
 	}
 
 	lndClient, err := lnd.NewLNDclient(lnd.LNDoptions{
 		Address:     c.LNDAddress,
 		MacaroonHex: c.LNDMacaroonHex,
 	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to initialize lnd service client: %w", err)
+		return nil, nil, nil, fmt.Errorf("failed to initialize lnd service client: %w", err)
 	}
 
-	return &service.LndhubService{
+	logger := lib.Logger(c.LogFilePath)
+	svc = &service.LndhubService{
 		Config:    c,
 		DB:        dbConn,
 		LndClient: lndClient,
-	}, nil
+		Logger:    logger,
+	}
+	getInfo, err := lndClient.GetInfo(ctx, &lnrpc.GetInfoRequest{})
+	if err != nil {
+		logger.Fatalf("Error getting node info: %v", err)
+	}
+	hexPubkey, err := hex.DecodeString(getInfo.IdentityPubkey)
+	if err != nil {
+		logger.Fatalf("Failed to decode IdentityPubkey: %v", err)
+	}
+	identityPubKey, err := btcec.ParsePubKey(hexPubkey[:], btcec.S256())
+	if err != nil {
+		logger.Fatalf("Failed to parse node IdentityPubkey: %v", err)
+	}
+	svc.IdentityPubkey = identityPubKey
+	logins = []controllers.CreateUserResponseBody{}
+	tokens = []string{}
+	for i := 0; i < usersToCreate; i++ {
```

Comment:
I feel like this function now does a lot of things. 
can't this user preparation be some function on some test service that we can split this up: first init and then create some users? 
We might not only need to create users but also invoices and other things in the future?

Reply:
It does do a lot of things, but creating users and fetching tokens is tested seperately.

Adding invoices can be done through [this function](https://github.com/getAlby/lndhub.go/blob/feature/integration-testing-payments/integration_tests/internal_payment_test.go#L76), which is used by multiple tests.

Also, you can't define a function on a receiver from another package. We can split it up but I think it's not needed as you'd want users in practically all of the tests.

Reply:
I've split up the function.

Reply:
👍